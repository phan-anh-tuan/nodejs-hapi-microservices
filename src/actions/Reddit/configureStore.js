import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { createLogger } from 'redux-logger';
import { combineReducers } from 'redux';
import { postsBySubreddit, selectedSubreddit} from './reducers.js'
import { todos, isFetchingTodos, visibilityFilters} from '../reducers.js'

const loggerMiddleware = createLogger();

const rootReducer = combineReducers({
    postsBySubreddit,
    selectedSubreddit,
    todos,
    isFetchingTodos,
    visibilityFilters
})

export default function configureStore(preloadedState) {
    return createStore(rootReducer,
                        preloadedState,
                        applyMiddleware(thunkMiddleware,
                                        loggerMiddleware
                        )
                     )
}
