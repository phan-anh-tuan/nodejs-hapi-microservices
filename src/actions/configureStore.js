import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { createLogger } from 'redux-logger';
import { combineReducers } from 'redux';
import { postsBySubreddit, selectedSubreddit} from './Reddit/reducers.js'
import { todos, isFetchingTodos, visibilityFilters} from './Todos/reducers.js'
import { resourceRequests } from './resource_requests/reducers.js'
import { VisibilityFilters } from './Todos/actionTypes.js'
const loggerMiddleware = createLogger();


const initialState = {
    postsBySubreddit: {
        reactjs: {
            isFetching: false,
            didInvalidate: true,
            items: []
        },
        frontend: {
            isFetching: false,
            didInvalidate: true,
            items: []
        }
    },
    selectedSubreddit: 'reactjs',
    todos: [],
    isFetchingTodos: false,
    visibilityFilters: VisibilityFilters.SHOW_ALL,
    resourceRequests: {
      isFetching: false,
      items: [],
      activeRequest: {
          isFetching: false,
          isSubmitting: false,
          showComment: false,
          data: {}
      },
      updatedAt: Date.now()
    }
}

const rootReducer = combineReducers({
    postsBySubreddit,
    selectedSubreddit,
    todos,
    isFetchingTodos,
    visibilityFilters,
    resourceRequests
})

export default function configureStore() {
    return createStore(rootReducer,
                        initialState,
                        applyMiddleware(thunkMiddleware,
                                        loggerMiddleware
                        )
                     )
}
