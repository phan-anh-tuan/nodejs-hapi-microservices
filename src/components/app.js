import React from 'react'
import Footer from './footer'
import AddTodo from '../containers/addTodo.js'
import VisibleTodoList from '../containers/visibleTodoList'

const App = ({ match }) => { 
  //console.log(`app.js ${JSON.stringify(match)}`);
  return (
    <div>
      <AddTodo />
      <VisibleTodoList filter={ (match.params && match.params.filter) || 'SHOW_ALL'}/>
      <Footer />
    </div>
)}


export default App