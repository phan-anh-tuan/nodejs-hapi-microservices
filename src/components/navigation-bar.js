import React from 'react'
import { Navbar, Nav, NavItem, Col, FormControl  } from 'react-bootstrap'

import PropTypes from 'prop-types'

class NavigationBar extends React.Component {
    render() {
        return (
            <Navbar collapseOnSelect fluid>
                <Navbar.Header>
                    <Navbar.Brand>
                    <a href="/"><img style={{ height: '32px', width: '32px', position: 'relative', top: '-9px'}} src="/assets/img/logo-square.png"/></a>
                    </Navbar.Brand>
                    <Navbar.Toggle />
                </Navbar.Header>
                { this.props.showSearchBox && <Col className='search-container' xs={8} lgHidden>
                    <FormControl type="text" placeholder="Enter some keyword..." name="search" className="searchbox-input-small"/>
                </Col>}
                <Navbar.Collapse>
                    {this.props.children}
                </Navbar.Collapse>
            </Navbar>
        );
    }
}
/*
NavigationBar.propTypes = {
    handleSubmit: PropTypes.func.isRequired
}
*/

export default NavigationBar