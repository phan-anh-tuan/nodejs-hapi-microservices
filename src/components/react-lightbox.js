import React from 'react'

// CSS from http://stackoverflow.com/questions/19064987/html-css-popup-div-on-text-click
// and http://stackoverflow.com/questions/10019797/pure-css-close-button
// source code from https://github.com/howtomakeaturn/React-Lightbox/blob/master/react-lightbox.jsx

export class LightboxModal extends React.Component{
    constructor(props) {
        super(props);
        this.whiteContentStyles = {
                                            position: 'fixed',
                                            top: '25%',
                                            left: '30%',
                                            right: '30%',
                                            backgroundColor: '#fff',
                                            color: '#7F7F7F',
                                            padding: '20px',
                                            border: '2px solid #ccc',
                                            borderRadius: '20px',
                                            boxShadow: '0 1px 5px #333',
                                            zIndex:'101'
                                        };
        this.blackOverlayStyles = {
                                            background: 'black',
                                            opacity: '.5',
                                            position: 'fixed',
                                            top: '0px',
                                            bottom: '0px',
                                            left: '0px',
                                            right: '0px',
                                            zIndex: '100'
                                        };
        this.closeTagStyles = {
                                        float: 'right',
                                        marginTop: '-30px',
                                        marginRight: '-30px',
                                        cursor: 'pointer',
                                        color: '#fff',
                                        border: '1px solid #AEAEAE',
                                        borderRadius: '30px',
                                        background: '#605F61',
                                        fontSize: '31px',
                                        fontWeight: 'bold',
                                        display: 'inline-block',
                                        lineHeight: '0px',
                                        padding: '11px 3px',
                                        textDecoration: 'none'
                                    };
        
    }
    

    

    

    componentDidMount(){
        document.addEventListener("keydown", function (e) {
            if ( (this.props.display) && (e.keyCode === 27) ){
                this.props.closeLightbox();
            }
        }.bind(this));
    }

    render(){

        const _lightboxModal = this;
        const {children, ...rest} = _lightboxModal.props;
        const childrenWithProps = React.Children.map(_lightboxModal.props.children,(child) => {
            const childWithProps = React.cloneElement(child, { ...rest})
            return childWithProps;
        });

        if (this.props.display){
            return (
                <div>
                    <div style={Object.assign({},this.blackOverlayStyles,!!this.props.blackOverlayStyles? this.props.blackOverlayStyles: {})} onClick={this.props.closeLightbox} />
                    <div style={Object.assign({},this.whiteContentStyles,!!this.props.whiteContentStyles? this.props.whiteContentStyles: {})}>
                        <a style={Object.assign({},this.closeTagStyles,!!this.props.closeTagStyles? this.props.closeTagStyles: {})} onClick={this.props.closeLightbox}>&times;</a>
                        {childrenWithProps}
                    </div>
            </div>
            );
        } else {
            return (<div></div>);
        }
    }
}


export class LightboxTrigger extends React.Component{
    render(){
        const _lightboxTrigger = this;
        const {children, ...rest} = _lightboxTrigger.props;
        const childrenWithProps = React.Children.map(_lightboxTrigger.props.children,(child) => {
            const childWithProps = React.cloneElement(child, { onClick: _lightboxTrigger.props.openLightbox, ...rest})
            return childWithProps;
        });
    
        return childrenWithProps[0];
    }
}


export class Lightbox extends React.Component {
    
    constructor(props) {
        super(props);
        this.openLightbox = this.openLightbox.bind(this);
        this.closeLightbox = this.closeLightbox.bind(this);
        this.setLightboxState = this.setLightboxState.bind(this);
        this.state = { display: false };
    }
    
    componentWillMount(){
        if (this.props.data)
            this.setState(this.props.data);
    }

    openLightbox(){
        this.setState({display: true});
    }

    closeLightbox() {
        this.setState({display: false});
    }

    setLightboxState(obj){
        this.setState(obj);
    }

    render(){
        const _lightbox = this;
        let i = 0;
        const childrenWithProps = React.Children.map(this.props.children,(child) => {
            
            i++;
            const childProps = {
                openLightbox: _lightbox.openLightbox,
                closeLightbox: _lightbox.closeLightbox,
                setLightboxState: _lightbox.setLightboxState,
                key: i
            };
            for (var j in _lightbox.state){
                childProps[j] = _lightbox.state[j];
            }
            const childWithProps = React.cloneElement(child, childProps);
            return childWithProps;
        });
    
        return (
            <div>
                {childrenWithProps}
            </div>
        );
    }
}

