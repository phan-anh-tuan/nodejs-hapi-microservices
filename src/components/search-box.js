import React from 'react'

class SearchBox extends React.Component {
    
    buttonUp(){
        
        var inputVal = $('.searchbox-input').val();
        inputVal = $.trim(inputVal).length;
        if( inputVal !== 0){
            $('.searchbox-icon').css('display','none');
        } else {
            $('.searchbox-input').val('');
            $('.searchbox-icon').css('display','block');
        }
    }
    componentDidMount() {
        
    }

    render() {
        return (<div className="searchbox-container">
            <form className="searchbox">
                <input type="search" placeholder="Enter some keyword..." name="search" className="searchbox-input" onKeyUp={this.buttonUp.bind(this)} required/>
                <input type="submit" className="searchbox-submit" value="GO" onClick={(e) => {e.preventDefault()}}/>
                <span className="searchbox-icon glyphicon glyphicon-search"></span>
            </form>
        </div>)
    }
}

export default SearchBox;
