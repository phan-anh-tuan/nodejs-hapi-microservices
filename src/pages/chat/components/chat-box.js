import React from 'react'

export default class ChatBox extends React.Component {

    constructor(props) {
        super(props)
        this.state = {}
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange(event) {
        const target = event.target;
        this.setState(Object.assign({}, this.state, { [target.id]: target.value }));
    }

    handleSubmit(e) {
        this.props.handleSubmit(this.props.id,this.state.m)
        //$(this.popupMessages).scrollTop($(this.popupMessages)[0].scrollHeight);
        this.setState(Object.assign({}, this.state, { m: '' }));
        e.preventDefault();
    }

    componentWillMount() {
        console.log(`chat/components/chat-box will mount`)
    }

    componentDidMount() {
        const uploadImageSelector = `div#${this.props.id} div.popup-head div.popup-head-right img:nth-child(2)`
        const videoCallImageSelector = `div#${this.props.id} div.popup-head div.popup-head-right img:nth-child(1)`
        const fileChooserSelector = `div#${this.props.id} input[type="file"]`
        const recipientId = this.props.id;
        const recipientName = this.props.name;
        $(uploadImageSelector).on("click", function() {
            $(fileChooserSelector).trigger("click");
        });
        $(videoCallImageSelector).on("click", function() {
            call(recipientName)
        });
        if(window.File && window.FileReader){ //These are the relevant HTML5 objects that we are going to use 
            $(fileChooserSelector).on('change', function(evnt){
                const SelectedFile = evnt.target.files[0];
                SelectedFiles.set(`${window.userId}:${recipientId}:${SelectedFile.name}`,SelectedFile);
                if(SelectedFile.name !== "")
                {
                    FReader = new FileReader();
                    FReader.onload = function(evnt){
                        //console.log('chat-box.js Emit Upload event')
                        window.socket.emit('fileupload:upload', { 'Name' : SelectedFile.name, Data : evnt.target.result, From: window.userId, FromUserName: window.username,  To: recipientId });
                    }
                    window.socket.emit('fileupload:start', { 'Name' : SelectedFile.name, 'Size' : SelectedFile.size, From: window.userId, To: recipientId});
                }
                else
                {
                    alert("Please Select A File");
                }
            });
        }
        else
        {
            alert("Your Browser Doesn't Support The File API Please Update Your Browser");
        }

        console.log(`chat/components/chat-box did mount`)
    }

    componentWillUnmount() {
        console.log(`chat/components/chat-box will unmount`)
    }

    render() {
          
        let _messages = [];
        this.props.messages.forEach( msg => {
            if (msg.type === 'url') {
                _messages.push(<li><strong>{msg.from.split(':')[0]}:</strong>&nbsp;<span dangerouslySetInnerHTML={{__html: msg.message}} /></li>)
            } else {
                _messages.push(<li><strong>{msg.from.split(':')[0]}:</strong>&nbsp;<span>{msg.message}</span></li>)
            }
        })
        return (<div className="popup-box chat-popup" id={this.props.id} style={this.props.right}>
                    <input type="file" style={{display: 'none'}} />
                    <div className="popup-head">
                        <div className="popup-head-left">{this.props.name}</div>
                        <div className="popup-head-right"><img width="30" height="30" src="/assets/img/videocall.png" />{' '}<img width="30" height="30" src="/assets/img/camera1600.png" />{'  '}<a onClick={() => this.props.handleClosePopup(this.props.id)}>&#10005;</a></div>
                        <div style={{clear: 'both'}}></div>
                    </div>
                    <div className="popup-messages"  ref={(input) => { this.popupMessages = input; }}>
                         <ul id="messages">{_messages}</ul>   
                    </div> 
                    <div className="popup-footer" style={{lineHeight: '2em'}}>
                        <form className='chat'>
                            <input id="m" autoComplete="off" onChange={this.handleChange} value={ this.state.m || ''} style={{width: '80%', marginTop: '1px'}}/>
                            <button onClick={this.handleSubmit} style={{fontWeight: 'bold', color: 'white', width: '20%', background: '#6d84b4', marginTop: '1px'}}>Send</button> 
                        </form>
                    </div>
                </div>)
    }
}  