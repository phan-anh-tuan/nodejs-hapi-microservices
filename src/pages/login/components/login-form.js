import React from 'react'
import { Row, Form, FormGroup, FormControl, Col, ControlLabel, Button, HelpBlock } from 'react-bootstrap'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
const Qs = require('qs')

import FieldGroup from '../../../components/form/field-group'
import { handleLoginRequest } from '../actions.js'


class LoginForm extends React.Component {
    constructor(props) {
        super(props);
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
        this.state = {};
    }

    handleChange(event) {
        const target = event.target;
        this.setState(Object.assign({}, this.state, { [target.id]: target.value }));
    }

    handleSubmit(e) {
        e.preventDefault();
        const { username, password } = this.state;                                                        
        this.props.handleSubmit({username, password})
                            .then( () => {  
                                if (!this.props.error) {
                                    const query = Qs.parse(window.location.search.substring(1));

                                    if (query.returnUrl) {
                                        window.location.href = query.returnUrl;
                                    }
                                    else {
                                        window.location.href = '/resource/request';
                                    }
                 
                                }
                            })
    }

    render() {
        const alerts = [];        
        alerts.push(<div key="danger" className="alert alert-danger">
                {this.props.error}
            </div>);
        return (
            <Form horizontal>
                {  
                    this.props.loading && 
                    <FormGroup>
                        <Col componentClass={ControlLabel} xsPush={1} xs={11} sm={2}>
                            {" "}
                        </Col>
                        <Col xsPush={1} xs={11} sm={6}>
                            <HelpBlock>{"Processing...."}</HelpBlock>
                        </Col>
                    </FormGroup>
                }
                { 
                    this.props.error && 
                      <FormGroup>
                        <Col xsPush={1} xs={11} smOffset={2} sm={6}>
                            {alerts}
                        </Col>
                    </FormGroup>
                }
                <FieldGroup id="username" label="Username:" type="text" placeholder="Username" onChange={this.handleChange} value={ this.state.username || ''} help={ this.props.help.username }/>
                <FieldGroup id="password" label="Password:" type="password" placeholder="Password" onChange={this.handleChange} value={ this.state.password || ''} help={ this.props.help.password}/>
                <FormGroup>
                    <Col xsPush={1} xs={11} smOffset={2} sm={6}>
                        <Button bsStyle="primary" type="submit" onClick={this.handleSubmit}>
                            Login
                        </Button>
                        <Link to="/signin/forgot" className="btn btn-link">Forgot your password?</Link>
                    </Col>
                </FormGroup>
            </Form>
        );
    }
}

LoginForm.propTypes = {
    handleSubmit: PropTypes.func.isRequired
}


const mapDispatchToProps = dispatch => {
  return {
    handleSubmit: (account) => {
        console.log(`login/components/login-form handle submission with parameters: ${JSON.stringify(account)}`);
        return dispatch(handleLoginRequest(account))
    }
  }
}

const mapStateToProps = (state,ownProps) => {
  return state.login;
}

export default connect(mapStateToProps,mapDispatchToProps)(LoginForm);