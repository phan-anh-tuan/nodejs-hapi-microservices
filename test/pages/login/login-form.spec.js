import React from 'react';
import { mount, shallow } from 'enzyme';
//import {expect} from 'chai';
//import jest from 'jest';

import sinon from 'sinon'
import { MemoryRouter } from 'react-router'
var fetchMock = require('fetch-mock');
import { Provider } from 'react-redux'
import LoginForm from '../../../src/pages/login/components/login-form'
import {LoginForm as Login} from '../../../src/pages/login/components/login-form'
//import store from '../../../src/pages/login/store'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import renderer from 'react-test-renderer'
import FieldGroup from '../../../src/components/form/field-group'

const { configure } = require('enzyme');
const Adapter  = require('enzyme-adapter-react-15');
configure({ adapter: new Adapter() });

/****************************************************************************
 * This code snippet below initialize jsdom jest test environment.
 * jsdom based test environment is similar to browser based test environment
 * which is required for enzyme mount
 * By default Jest default test environment is jsdom however you can override 
 * this setting in package.json or in jest.config.js which is the case here
 ****************************************************************************/
const { JSDOM } = require('jsdom');
const jsdom = new JSDOM('<!doctype html><html><body></body></html>');
const { window } = jsdom;

function copyProps(src, target) {
  const props = Object.getOwnPropertyNames(src)
    .filter(prop => typeof target[prop] === 'undefined')
    .map(prop => Object.getOwnPropertyDescriptor(src, prop));
  Object.defineProperties(target, props);
}

global.window = window;
global.document = window.document;
global.navigator = {
  userAgent: 'node.js',
};

beforeAll(()=>
{
    copyProps(window, global);
})
/******************************************
 * end of jest test environment setting up
 * now you can use enzyme mount
 ******************************************/

describe('<LoginForm/>', function () {
  let _store;
  let wrapper;
  beforeEach(function(done){
        //this.server = sinon.createFakeServer();
        //this.xhr = sinon.useFakeXMLHttpRequest();
        //console.log(`xhr ${window.xhr}`)
        console.log('BeforeEach inside login-form.spec.js')
        fetchMock.mock('http://localhost:3000/api/login', { 'data': 'hello'});
        
        const middlewares = [thunk] // add your middlewares like `redux-thunk`
        const mockStore = configureStore(middlewares)
        _store = mockStore(
          {
            login: {
              loading: false,
              success: false,
              error: undefined,
              hasError: {},
              help: {}
            }
          })
        done();
  })
  
  afterEach(function(done){
        //this.server.restore();
        fetchMock.restore();
        if (wrapper.unmount) wrapper.unmount();
        done();
  })
  it('should call handleSubmit once', function () {
  
    /*this.server.respondWith("POST", "/api/login",
    [200, { "Content-Type": "application/json" },
     '{}']);*/

    //const myMock = fetchMock.sandbox().mock('http://localhost:3000/api/login', { 'data': 'hello'});

    //fetch('http://localhost:3000/api/login')
    //.then( response => response.json())
    //.then( data => console.log(data))
    //.catch( error => console.log(error))
    //expect(store).to.be.a('function')

    //var xhr = sinon.spy(global.window, "XMLHttpRequest"); 
    //const wrapper = mount(<Provider store={store()}><MemoryRouter><LoginForm/></MemoryRouter></Provider>);
    wrapper = mount(<Provider store={_store}><MemoryRouter><LoginForm/></MemoryRouter></Provider>);
    expect(wrapper.find('button').length).toBe(1);
    wrapper.find('button').simulate('click');
    //expect(mockCallback.mock.calls.length).toBe(1);
    //expect(mockCallback.calledOnce).toBeTruthy()
    //expect(myMock.called()).to.be.true;
    //expect(fetchMock.called()).to.be.true;
    const actions = _store.getActions()
    console.log('actions',actions)
    //expect(actions[0].type).to.equal('LOGIN')
  });

  it('calls componentDidMount', function () {
    //sinon.spy(LoginForm.prototype, 'componentDidMount');
    const spy = jest.spyOn(LoginForm.prototype, 'componentDidMount');
    wrapper = mount(<Provider store={_store}><MemoryRouter><LoginForm/></MemoryRouter></Provider>);
    //expect(LoginForm.prototype.componentDidMount.calledOnce).toBeTruthy()   
    expect(spy).toHaveBeenCalled();
    spy.mockReset();
    spy.mockRestore();
  });

  it('+++capturing Snapshot of Home', () => {
    const mockCallback = jest.fn();
    const renderedValue =  renderer.create(<Provider store={_store}><MemoryRouter><LoginForm/></MemoryRouter></Provider>).toJSON()
    expect(renderedValue).toMatchSnapshot()
    //jest.expect(renderedValue).toMatchSnapshot();
  });
/*
  it('should have props for email and src', function () {
    const wrapper = shallow(<Avatar email='someone@example.com' src='http://placehold.it/200x200'/>);
    console.log(JSON.stringify(wrapper.props()))
    expect(wrapper.instance().props.email).to.be.a('string').that.equal('someone@example.com');
    expect(wrapper.instance().props.src).to.be.a('string').that.equal('http://placehold.it/200x200');
  });*/
});


describe('>>>L O G I N --- Shallow Render REACT COMPONENTS',()=>{
  let wrapper

  beforeEach(()=>{
    const mockCallback = jest.fn();
      wrapper = shallow(<Login/>)
  }) 

  it('+++ render the DUMB component', () => {
    
     //expect(wrapper.length).to.equal(1)
     //wrapper.children().forEach(node=> console.log(node.debug()))
     //expect(wrapper.find(FieldGroup).exists()).toBeTruthy();
     console.log('Login Form debug() ',wrapper.debug());
     console.log('FieldGroup html() ',wrapper.find(FieldGroup).at(0).html());
  });
  
});



