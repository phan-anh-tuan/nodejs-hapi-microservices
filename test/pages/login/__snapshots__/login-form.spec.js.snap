// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm/> +++capturing Snapshot of Home 1`] = `
<form
  className="form-horizontal"
>
  <div
    className="form-group"
  >
    <label
      className="col-sm-2 col-xs-11 col-xs-push-1 control-label"
      htmlFor="username"
    >
      Username:
    </label>
    <div
      className="col-sm-6 col-xs-10 col-xs-push-1"
    >
      <input
        className="form-control"
        id="username"
        onChange={[Function]}
        placeholder="Username"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      className="col-sm-2 col-xs-11 col-xs-push-1 control-label"
      htmlFor="password"
    >
      Password:
    </label>
    <div
      className="col-sm-6 col-xs-10 col-xs-push-1"
    >
      <input
        className="form-control"
        id="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <div
      className="col-sm-6 col-sm-offset-2 col-xs-11 col-xs-push-1"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="submit"
      >
        Login
      </button>
      <a
        className="btn btn-link"
        href="/signin/forgot"
        onClick={[Function]}
      >
        Forgot your password?
      </a>
    </div>
  </div>
</form>
`;
